qiskit/transpiler/passes/synthesis/high_level_synthesis.py

```diff
@@ -131,24 +150,45 @@ def run(self, dag: DAGCircuit) -> DAGCircuit:
                 methods = []
 
             for method in methods:
-                plugin_name, plugin_args = method
 
-                if plugin_name not in hls_plugin_manager.method_names(node.name):
-                    raise TranspilerError(
-                        "Specified method: %s not found in available plugins for %s"
-                        % (plugin_name, node.name)
-                    )
+                # There are two ways to specify an individual method being run, either a tuple
+                #   ("kms", {"all_mats": 1, "max_paths": 100, "orig_circuit": 0}),
+                # or as a class instance
+                #   KMSSynthesisLinearFunction(all_mats=1, max_paths=100, orig_circuit=0).
+                if isinstance(method, tuple):
+                    plugin_name, plugin_args = method
+
+                    if plugin_name not in hls_plugin_manager.method_names(node.name):
+                        raise TranspilerError(
+                            "Specified method: %s not found in available plugins for %s"
+                            % (plugin_name, node.name)
+                        )
```

Comment:
Can we do this in a standalone PR, it's really a separate logical API change. I think it's a good change to make but I worry about bundling so much in a single PR.

Reply:
Yes, this was done in https://github.com/Qiskit/qiskit-terra/pull/9413.